container.go

```diff
@@ -35,8 +40,14 @@ type Container interface {
 	SessionID() string                                              // get session id
 	Start(context.Context) error                                    // start the container
 	Terminate(context.Context) error                                // terminate the container
+	Stop(context.Context) error                                     // stop the container
+	Remove(context.Context, bool) error                             // remove the container
 	Logs(context.Context) (io.ReadCloser, error)                    // Get logs of the container
 	Name(context.Context) (string, error)                           // get container name
+	IsRunning(ctx context.Context) (bool, error)                    // is state of container 'running'
+	State(ctx context.Context) (*types.ContainerState, error)       // state of container
+	Image(context.Context) (string, error)                          // get container image
+	ResetCache(context.Context)                                     // reset internal testcontainers-go cache
```

Comment:
`ResetCache` should be a private function, it does not need to be in the Container interface. We can implement it only for the docker provider.

Thanks